CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/fragments/CameraFragment.kt

```diff
@@ -99,7 +92,7 @@ class CameraFragment : Fragment() {
     private var preview: Preview? = null
     private var imageCapture: ImageCapture? = null
     private var imageAnalyzer: ImageAnalysis? = null
-    private var camera: Camera? = null
+    private var cameraProvider: ProcessCameraProvider? = null
```

Comment:
I'd like to keep a reference to `Camera` since it will be handy when we add some of the camera controls like torch. Can we avoid adding a reference to `ProcessCameraProvider` by passing it as an argument from `setupCamera` to `bindCameraUseCases`?

Reply:
I'll add the `Camera` reference back, I removed it since it wasn't being used.

Regarding the reference to `ProcessCameraProvider`, it's needed for when `onConfigurationChanged()` is called.

Previously, the flow was:
1. Fragment is first created.
2. lensFacing is set to a back camera.
3. Two things happen:
3.1. CameraX initialization (`cameraProviderFuture ...`)
3.2. `bindtoLifecycle()` binds the use cases.

This CL separates (3.1) and (3.2), and inverts (2) and (3.1). So when `onConfigurationChanged()` is called, a `ProcessCameraProvider` has to have been stored from (3.1).